src/libraries/Common/tests/System/Net/Http/Http3LoopbackConnection.cs

```diff
@@ -188,11 +204,17 @@ public async Task<(Http3LoopbackStream clientControlStream, Http3LoopbackStream
             return (controlStream, requestStream);
         }
 
-        public async Task EstablishControlStreamAsync()
+        public async Task EstablishControlStreamAsync(ICollection<(long settingId, long settingValue)> settings = null)
         {
             _outboundControlStream = await OpenUnidirectionalStreamAsync();
             await _outboundControlStream.SendUnidirectionalStreamTypeAsync(Http3LoopbackStream.ControlStream);
-            await _outboundControlStream.SendSettingsFrameAsync();
+            await _outboundControlStream.SendSettingsFrameAsync(settings);
+        }
+
+        public async Task<QuicStream> OpenWebtransportStreamAsync(QuicStreamType type)
+        {
+            QuicStream stream = await _connection.OpenOutboundStreamAsync(type);
+            return stream;
         }
```

Comment:
Should this method also send the required header on the stream, for it to qualify being a webtransport stream?

Reply:
Yes, but the header frame is sent inside the tests file(I have an aditional function which does that)
The Http3LoopbackConnection file does not handle sending stream stuff, this part being handled in the loopback Stream usually
And I did not believe calling a static function from my tests which handles sending the header inside this file would be the best solution for keeping the connection file purpose
I can do inside the tests file a function which calls this function along with the one that send the header, but I do not think sending stuff on quic inside this file is a good idea

Reply:
Ok. Then maybe a better name for this function would be "OpenRawQuicStream" or something like that. Because this stream right now is not tied to webtransport at all.

Then you can have a function in your test file, that does something like

```c#
public static async Task<QuicStream> OpenWebtransportStreamAsync(Http3LoopbackConnection connection, QuicStreamType type, long sessionId)
{
    QuicStream stream = await connection.OpenRawQuicStream(type);
    await SendWebtransportStreamHeaderAsync(
        stream,
        type == QuicStreamType.Unidirectional ? Http3LoopbackStream.UnidirectionalWebtransportStream : Http3LoopbackStream.BidirectionalWebtransportStream,
        sessionId);
    return stream;
}
```
so each test wouldn't have to repeat that everywhere.

(I re-read your message above and that's exactly what you were suggesting 😄)